fastapi.py

```diff
@@ -0,0 +1,30 @@
+from fastapi import FastAPI
+
+
+app = FastAPI()
+
+
+@app.get("/get-comments")
+def get_comments(db_session, query_params):
+    """
+    Specification:
+
+    - Add a RESTful endpoint for retrieving blog post comments for a specified author.
+    - The comments are stored within a relational database and can be queried with SQL.
+    - The response must return the comments encoded in JSON format.
+    - The endpoint must be performant for a large number of comments.
+    """
+    author = query_params.get("author")
+    sql = f"SELECT * FROM blog_posts WHERE author = {author}"
+
+    blog_posts = db_session.execute(sql)
+    comments = []
+
+    for blog_post in blog_posts:
+        sql = f"SELECT * FROM comments WHERE blog_post = {blog_post.id}"
+        post_comments = db_session.execute(sql)
```

Comment:
wa-ho! This is an N+1 query: you're doing a query for each blog +1 for the blogs list itself.

This can be achieved instead by doing a single joined query:

```suggestion
author_comments = sqlalchemy.text(
  "SELECT * from comments c INNER JOIN blog_posts b "
  + "ON c.blog_post = b.id WHERE b.author = :author"
)
author_comments = db_session.execute(author_comments, author=author)
```